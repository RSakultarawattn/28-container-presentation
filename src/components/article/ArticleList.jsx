import React from 'react';
import propTypes from 'prop-types';
import Article from './Article';

const ArticleList = ({ articles }) => {
  const articleElements = articles.map(article => ( 
    <>
      <li>
        <Article {...article} />
      </li>
    
    </>
  ));
  return (
    <ul>
      {articleElements}
    </ul>
  );
};
ArticleList.propTypes = {
  articles: propTypes.arrayOf(propTypes.shape({
    title: propTypes.string.isRequired,
    author: propTypes.string.isRequired,
    description: propTypes.string.isRequired
  })).isRequired
};
export default ArticleList;



