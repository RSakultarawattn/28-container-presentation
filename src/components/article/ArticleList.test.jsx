import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ArticleList from './ArticleList';

describe('Article item component', () => {
  afterEach(() => cleanup());
  it('renders an article', () => {
    const { asFragment } = render(<ArticleList
      articles={[]} 
    />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
