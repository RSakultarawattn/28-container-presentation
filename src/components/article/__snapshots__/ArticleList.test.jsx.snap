// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article item component renders an article 1`] = `
<DocumentFragment>
  <h1>
    Make a search....
  </h1>
</DocumentFragment>
`;
